zou/app/blueprints/crud/task.py

```diff
@@ -46,6 +50,19 @@ def post(self):
             is_assignees = "assignees" in data
             assignees = None
 
+            task_type = tasks_service.get_task_type(data["task_type_id"])
+            entity = entities_service.get_entity(data["entity_id"])
+            if task_type["for_entity"] == "Asset":
+                if not assets_service.is_asset_dict(entity):
+                    raise WrongTaskTypeForEntityException
+            elif (
+                entities_service.get_temporal_entity_type_by_name(
+                    task_type["for_entity"]
+                )["id"]
+                != entity["entity_type_id"]
+            ):
+                raise WrongTaskTypeForEntityException
```

Comment:
You should handle this error the same way other errors are managed: 
https://github.com/cgwire/zou/blob/master/zou/app/__init__.py#L81-L83